src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -4051,6 +4051,119 @@ .maxstack  3
 ");
         }
 
+        [Fact]
+        public void DefaultOfFunctionPointerType()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+using System;
+unsafe class C
+{
+    static void Main()
+    {
+        delegate*<void> ptr = default;
+        Console.Write(ptr is null);
+    }
+}", expectedOutput: "True");
+
+            verifier.VerifyIL("C.Main", @"
+{
+  // Code size       11 (0xb)
+  .maxstack  2
+  IL_0000:  ldc.i4.0
+  IL_0001:  conv.u
+  IL_0002:  ldnull
+  IL_0003:  ceq
+  IL_0005:  call       ""void System.Console.Write(bool)""
+  IL_000a:  ret
+}
+");
+        }
+
+        [Fact]
+        public void ParamsArrayOfFunctionPointers()
+        {
+            var verifier = CompileAndVerifyFunctionPointers(@"
+unsafe class C
+{
+    static void Params(params delegate*<void>[] funcs)
+    {
+        foreach (var f in funcs)
+        {
+            f();
+        }
+    }
+
+    static void Main()
+    {
+        Params();
+    }
+}", expectedOutput: "");
+
+            verifier.VerifyIL("C.Main", expectedIL: @"
+{
+  // Code size       12 (0xc)
+  .maxstack  1
+  IL_0000:  ldc.i4.0
+  IL_0001:  newarr     ""delegate*<void>""
+  IL_0006:  call       ""void C.Params(params delegate*<void>[])""
+  IL_000b:  ret
+}
+");
+        }
+
+        [Fact]
+        public void SpanOfFunctionPointers()
```

Comment:
>SpanOfFunctionPointers [](start = 20, length = 22)

Would it make sense to test a scenario where the Span is used explicitly?

Reply:
I can add a test to verify that we correctly error on using it as a type parameter.

---
In reply to: [407735110](https://github.com/dotnet/roslyn/pull/43248#discussion_r407735110) [](ancestors = 407735110)